src/hotspot/share/utilities/exceptions.cpp

```diff
@@ -609,3 +620,74 @@ void Exceptions::log_exception(Handle exception, const char* message) {
                          MAX_LEN, message);
   }
 }
+
+// This is called from InterpreterRuntime::exception_handler_for_exception(), which is the only
+// easy way to be notified in the VM that an _athrow bytecode has been executed. (The alternative
+// would be to add hooks into the interpreter and compiler, for all platforms ...).
+//
+// Unfortunately, InterpreterRuntime::exception_handler_for_exception() is called for every level
+// of the Java stack when looking for an exception handler. To avoid excessive output,
+// we print the stack only when the bci points to an _athrow bytecode.
+//
+// NOTE: exceptions that are NOT thrown by _athrow are handled by Exceptions::special_exception()
+// and Exceptions::_throw()).
+void Exceptions::log_exception_stacktrace(Handle exception, methodHandle method, int bci) {
+  if (method->is_native() || (Bytecodes::Code) *method->bcp_from(bci) == Bytecodes::_athrow) {
```

Comment:
Do you mean !method->is_native() && athrow?  How do you get here from a native method ?

Reply:
I fixed it. I thought we could get there with a native method, but looking at `InterpreterRuntime::exception_handler_for_exception()`, we can't.

Anyway, I left the `!method->is_native()` there just for safety.